### ISSUE TITLE
When using SourceMap, it gets very laggy

### ISSUE THREAD

**Issue by taichi-jp** [react-native-community/react-native-tab-view#584]
If I use SourceMap like this;
```js
<TabViewAnimated
        navigationState={this.state}
        renderScene={SourceMap({
            foo: FooView,
            bar: BarView,
            fooBar: FooBarView,
        })}
        renderHeader={renderTabBar}
        onIndexChange={index => this.setState({ index })}
/>
```

```js
this.state = {
    index: 0,
    routes: [
      { key: 'foo', title: 'FOO' },
      { key: 'bar', title: 'BAR' },
      { key: 'fooBar', title: 'FOO_BAR' },
    ],
  };

const renderTabBar = () => (
     <TabBar
          {...props}
     />
);
```
Changing scenes by tapping or swiping TabBar gets *very* laggy in my iPhone6(iOS 11) and Android phone(Xperia XZ Premium with Android8).

I guess the codes in https://github.com/react-native-community/react-native-tab-view/blob/master/src/SceneMap.js is not very fast.

On the other hand, when I implement `renderScene` like this, it works well.

```js
const renderScene = ({ route }) => {
    switch(route.key) {
        case 'foo':
            return Foo;
        ...
    }
}
```

**Comment by gabrielreiscom** [comment]
I'm experiencing a similar issue on iPhone 6

**Comment by oisu** [comment]
So am I only on Android. I got stuck the laggy issue for a while but now it's solved after just replacing `renderScene`. Thanks for saving my time.

**Comment by satya164** [comment]
Can you share the full code so I can repro this? It's strange that using `SceneMap` will be slower because using it optimizes the scenes to prevent re-renders.

**Comment by taichi-jp** [comment]
@satya164 
Unfortunately I cannot share the code that got laggy in my environment becuase it's the code of my company.

In https://github.com/react-native-community/react-native-tab-view/issues/584#issue-350749120, actually `FooView/BarView/FooBarView` are big components, and connected to redux using `react-redux` .

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
